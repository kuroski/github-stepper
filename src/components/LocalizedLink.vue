<template>
  <router-link
    :to="toWithLocale"
    v-bind="$attrs"
    :class="{
      btn: ['primary', 'secondary'].includes(type),
      'btn--primary': type === 'primary',
      'btn--secondary': type === 'secondary'
    }"
  >
    <slot />
  </router-link>
</template>

<script>
export default {
  name: "LocalizedLink",
  props: {
    to: {
      type: Object,
      required: true
    },
    type: {
      type: String,
      default: "link",
      validator(value) {
        return ["link", "primary", "secondary"].includes(value);
      }
    }
  },
  computed: {
    toWithLocale() {
      const params = this.to.params || {};
      return {
        ...this.to,
        params: {
          ...params,
          locale: this.$i18n.locale
        }
      };
    }
  }
};
</script>
