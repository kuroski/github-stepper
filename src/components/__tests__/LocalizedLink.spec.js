import { renderWithDependencies } from "@unit/helpers";
import LocalizedLink from "@/components/LocalizedLink";

describe("LocalizedLink", () => {
  it("renders a router-link with current system locale", () => {
    const { getByText } = renderWithDependencies(LocalizedLink, {
      props: {
        to: { name: "home" }
      },
      slots: {
        default: "My link"
      }
    });

    const link = getByText("My link");
    expect(link).toMatchSnapshot();
    expect(link).toHaveAttribute("href", "#/en/");
  });
});
