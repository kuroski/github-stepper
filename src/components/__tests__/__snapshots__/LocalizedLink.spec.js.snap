// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalizedLink renders a router-link with current system locale 1`] = `
<a
  class=""
  href="#/en/"
>
  My link
</a>
`;
